drivers/adc/ad405x/ad405x.c

```diff
@@ -0,0 +1,900 @@
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad405x_disable_invert_on_chop(struct ad405x_dev *dev)
+{
+	int ret;
+
+	if (!dev)
+		return - EINVAL;
+
+	ret = ad405x_update_bits(dev,
+				 AD405X_REG_GPIO_CONFIG,
+				 AD405X_INVERT_ON_CHOP_MSK,
+				 no_os_field_prep(AD405X_INVERT_ON_CHOP_MSK, AD405X_INVERT_ON_CHOP_DISABLED));
+	if (ret)
+		return ret;
+
+	dev->invert_on_chop_status = AD405X_INVERT_ON_CHOP_DISABLED;
+
+	return 0;
+}
+
+/**
+ * @brief Get INVERT_ON_CHOP setting
+ * @param dev - The device structure.
+ * @return INVERT_ON_CHOP setting
+ */
+enum ad405x_invert_on_chop ad405x_get_invert_on_chop_state(
+	struct ad405x_dev *dev)
+{
+	if (!dev)
+		return - EINVAL;
+
+	return dev->invert_on_chop_status;
+}
+
+/**
+ * @brief Set GP mode
+ * @param dev - The device structure.
+ * @param gp - GP1/GP0 select.
+ * @param mode - mode select.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad405x_set_gp_mode(struct ad405x_dev *dev,
+		       enum ad405x_gp_select gp,
+		       enum ad405x_gp_mode mode)
+{
+	int ret;
+	unsigned char msk;
+
+	if (!dev)
+		return - EINVAL;
+
+	switch(gp) {
+	case AD405X_GP_0 :
+		msk = AD405X_GP0_MODE_MSK;
+		break;
+	case AD405X_GP_1 :
+		msk = AD405X_GP1_MODE_MSK;
+		break;
+	default :
+		return - EINVAL;
+	}
+
+	if (gp == AD405X_GP_0 && mode == AD405X_GP_MODE_DEV_RDY)
+		return - EINVAL;
+
+	ret = ad405x_update_bits(dev,
+				 AD405X_REG_GPIO_CONFIG,
+				 msk,
+				 no_os_field_prep(msk, mode));
+	if (ret)
+		return ret;
+
+	switch(gp) {
+	case AD405X_GP_0 :
+		dev->gp0_mode = mode;
+		break;
+	case AD405X_GP_1 :
+		dev->gp1_mode = mode;
+		break;
+	default :
+		return - EINVAL;
+	}
+
+	return 0;
+}
+
+/** G */
+/**
+ * @brief Get GP mode setting
+ * @param dev - The device structure.
+ * @param gp - GP1/GP0 select.
+ * @return GP mode.
+ */
+enum ad405x_gp_mode ad405x_get_gp_mode(
+	struct ad405x_dev *dev,
+	enum ad405x_gp_select gp)
+{
+	if (!dev)
+		return - EINVAL;
+
+	switch(gp) {
+	case AD405X_GP_0 :
+		return dev->gp0_mode;
+	case AD405X_GP_1 :
+		return dev->gp1_mode;
+	default :
+		return - EINVAL;
+	}
+}
+
+/**
+ * @brief Set output data format.
+ * @param dev - The device structure.
+ * @param data_format - format select.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad405x_set_data_format(struct ad405x_dev *dev,
+			   enum ad405x_out_data_format data_format)
+{
+	int ret;
+
+	if (!dev)
+		return - EINVAL;
+
+	ret = ad405x_update_bits(dev,
+				 AD405X_REG_ADC_MODES,
+				 AD405X_ADC_DATA_FORMAT_MSK,
+				 no_os_field_prep(AD405X_ADC_DATA_FORMAT_MSK, data_format));
+	if (ret)
+		return ret;
+
+	dev->data_format = data_format;
+
+	return 0;
+}
+
+/**
+ * @brief Get output data format setting.
+ * @param dev - The device structure.
+ * @return DATA_FORMAT setting or negative error code.
+ */
+enum ad405x_out_data_format ad405x_get_data_format(
+	struct ad405x_dev *dev)
```

Comment:
same here: is this function really needed?

Reply:
Not needed. Will remove